code/datums/repositories/crew.dm

```diff
@@ -24,7 +24,8 @@ var/global/datum/repository/crew/crew_repository = new()
 	var/tracked = scan()
 	for(var/obj/item/clothing/under/C in tracked)
 		var/turf/pos = get_turf(C)
-		if((C) && (C.has_sensor) && (pos) && (pos.z == zLevel) && (C.sensor_mode != SUIT_SENSOR_OFF) && !(is_jammed(C)))
+		var/area/B = get_area(pos) //VOREStation Add: No sensor in Dorm
+		if((C) && (C.has_sensor) && (pos) && (pos.z == zLevel) && (C.sensor_mode != SUIT_SENSOR_OFF) && !(is_jammed(C)) && !(B.block_suit_sensors)) //VOREStation Edit

```

Comment:
Sorry I didn't notice this before, this whole section could be done better. Mind touching it up while you're in there?

`var/area/B = pos?.loc` works, can save a get_area() call because the turf's loc is always the area it's in (and the nullsafe operator in case the turf is null)

The big if() chain can have a couple optimizations to just be...

`if((C.has_sensor) && (pos?.z == zLevel) && (C.sensor_mode != SUIT_SENSOR_OFF) && !(B.block_suit_sensors) && !(is_jammed(C)))`

C will definitely exist, there's no potential for pausing between getting the list of clothing/unders and iterating over it (maybe there should be...). Pos only has one thing accessed on it, can just use nullsafe dereference there. I put the block suit sensors test before jammed test because I think it's more likely to be in dorms than to be bluespace jammed, and more likely things should come earlier... could probably be optimized more, maybe C.has_sensor is needless, iono